pageserver/src/tenant/mgr.rs

```diff
@@ -262,14 +268,31 @@ pub async fn shutdown_all_tenants() {
     for (tenant_id, tenant) in tenants_to_shut_down {
         join_set.spawn(
             async move {
+                let (_guard, shutdown_progress) = completion::channel();
                 let freeze_and_flush = true;
+                if let Err(progress) = tenant.shutdown(shutdown_progress, freeze_and_flush).await {
+                    drop(_guard);
+                    // there is already something else shutting down the tenant (detach, ignore), lets wait for it
+                    let mut remaining = std::pin::pin!(progress.wait());
+                    let remaining = tokio::select! {
+                        biased;
+                        _ = &mut remaining => {
+                            None
+                        },
+                        _ = tokio::time::sleep(std::time::Duration::from_secs(5)) => {
+                            // in practice we might not have a lot to go, since systemd is going to
+                            // SIGKILL us at 10s, but we can try. delete tenant might take a while,
+                            // so put out a warning.
+                            warn!("waiting for the other shutdown to complete");
+                            Some(remaining)
+                        }
+                    };
 
-                match tenant.shutdown(freeze_and_flush).await {
-                    Ok(()) => debug!("tenant successfully stopped"),
-                    Err(super::ShutdownError::AlreadyStopping) => {
-                        warn!("tenant was already shutting down")
+                    if let Some(remaining) = remaining {
+                        remaining.await;
                     }
```

Comment:
```suggestion
                    tokio::select! {
                        biased;
                        _ = &mut remaining => {
                            None
                        },
                        _ = tokio::time::sleep(std::time::Duration::from_secs(5)) => {
                            // in practice we might not have a lot to go, since systemd is going to
                            // SIGKILL us at 10s, but we can try. delete tenant might take a while,
                            // so put out a warning.
                            warn!("shutdown coalesced with a previous shutdown is taking longer than 5 seconds");
                            remaining.await;
                        }
                    };

```

Reply:
I don't understand the suggestion, did I mess up the formatting?

Reply:
Aah maybe it suggests to move the remaining into the arm with the logging, while keeping the None in the other one.

Reply:
Moved it into the sleep arm regardless in 96c6fb541.